src/SeleniumTestCase.php

```diff
@@ -3,51 +3,65 @@
 namespace Modelizer\Selenium;
 
 use Modelizer\Selenium\Services\Application as Laravel;
+use Modelizer\Selenium\Services\InteractsWithPage as Interactions;
+use Modelizer\Selenium\Services\WaitsForElements;
+use Modelizer\Selenium\Services\WorksWithDatabases;
 use PHPUnit_Extensions_Selenium2TestCase;
 
 class SeleniumTestCase extends PHPUnit_Extensions_Selenium2TestCase
 {
-    use Laravel;
+    use Laravel,
+        Interactions,
+        WorksWithDatabases,
+        WaitsForElements;
+
+    /**
+     * @var string
+     */
+    protected $baseUrl;
 
     protected function setUp()
     {
         $this->setUpLaravel();
-
-        $this->setBrowserUrl(env('APP_URL', 'http://localhost/'));
+        $this->baseUrl = env('APP_URL', 'http://localhost/');
+        $this->setBrowserUrl($this->baseUrl);
         $this->setBrowser(env('DEFAULT_BROWSER', 'chrome'));
     }
 
-    protected function visit($path)
-    {
-        $this->url($path);
-
-        return $this;
-    }
-
-    protected function see($text, $tag = 'html')
+    public function setupPage()
     {
-        $this->assertContains($text, $this->byTag($tag)->text());
-
-        return $this;
+        if (!empty(getenv('SELENIUM_WIDTH')) && !empty(getenv('SELENIUM_HEIGHT')) &&
+            is_int(intval(getenv('SELENIUM_WIDTH'))) && is_int(intval(getenv('SELENIUM_HEIGHT')))) {
+            $this->prepareSession()->currentWindow()->size([
+                'width'  => intval(getenv('SELENIUM_WIDTH')),
+                'height' => intval(getenv('SELENIUM_HEIGHT')),
+            ]);
+        }
     }
 
-    protected function hold($seconds)
+    /**
+     * Force selenium to wait.
+     *
+     * @param int|float $seconds The number of seconds or partial seconds to wait
+     *
+     * @return $this
+     */
+    protected function wait($seconds = 1)
     {
-        sleep($seconds);
+        usleep($seconds * 1000000);
 
         return $this;
     }
 
-    protected function submitForm($inputs, $selector)
+    /**
+     * Alias for wait.
+     *
+     * @param int $seconds
+     *
+     * @return SeleniumTestCase
+     */
+    public function hold($seconds = 1)
     {
-        $form = $this->byCssSelector($selector);
-
-        foreach ($inputs as $input => $value) {
-            $form->byName($input)->value($value);
-        }
-
-        $form->submit();
-
-        return $this;
+        return $this->wait($seconds);
```

Comment:
Wait and hold are the same thing, even if you're waiting for an AJAX request.

If you're waiting to see the result of something from AJAX like say your HTML table gets populated or shown from data, then I'd use the "waitsForElementBy" syntax as that is the end result of an AJAX query.

This can be done for things like waiting for a Sweet Alert alert to show up, or a notification box showing up if you save records with AJAX.